packages/contentstack-export-to-csv/test/unit/commands/export-to-csv.test.js

```diff
@@ -1,219 +1,293 @@
+      })
+      .command(['cm:export-to-csv'])
+      .it('Entries CSV file should be created with prompt');
   });
+});
 
-  it('Should show an error that user does not have org permissions to perform the operation if user enters such org', async () => {
-    const args = ['--action', 'users', '--org', process.env.ORG_WITH_NO_PERMISSION];
-    await ExportToCsvCommand.run(args);
-    assert.calledWith(errorStub, `You don't have the permission to do this operation.`);
+describe('export-to-csv with action users', () => {
+  describe('Export users csv file with flags', () => {
+    test
+      .stdout({ print: process.env.PRINT === 'true' || false })
+      .stub(fs, 'createWriteStream', () => new PassThrough())
+      .stub(mkdirp, 'sync', () => {})
+      .stub(process, 'chdir', () => {})
+      .nock(cma, (api) => {
+        api.get('/v3/user?include_orgs_roles=true').reply(200, { user: mockData.users[0] });
+      })
+      .nock(cma, (api) => {
+        api.get(`/v3/organizations/${mockData.organizations[0].uid}/roles`).reply(200, { roles: mockData.roles });
+      })
+      .nock(cma, (api) => {
+        api.get('/v3/user?include_orgs_roles=true').reply(200, { user: mockData.users[0] });
+      })
+      .nock(cma, (api) => {
+        api
+          .get(`/v3/organizations/${mockData.organizations[0].uid}/share?skip=0&page=1&limit=100`)
+          .reply(200, { users: mockData.users });
+      })
+      .command(['cm:export-to-csv', '--action', 'users', '--org', mockData.organizations[0].uid])
+      .it('Users csv file should be successfully created');
   });
 
-  it('Should create a file starting with the name passed as org-name flag', async () => {
-    const args = ['--action', 'users', '--org', process.env.ORG, '--org-name', 'okok'];
-    await ExportToCsvCommand.run(args);
-    assert.calledWith(consoleLogStub, `Writing organization details to file: ${process.cwd()}/okok_users_export.csv`);
+  describe('Export users csv file with prompt', () => {
+    test
+      .stdout({ print: process.env.PRINT === 'true' || false })
+      .stub(fs, 'createWriteStream', () => new PassThrough())
+      .stub(mkdirp, 'sync', () => {})
+      .stub(process, 'chdir', () => {})
+      .stub(inquirer, 'registerPrompt', () => {})
+      .stub(inquirer, 'prompt', () => {
+        return Promise.resolve({ action: 'users', chosenOrg: mockData.organizations[0].name });
+      })
+      .nock(cma, (api) => {
+        api.get(`/v3/organizations?limit=100`).reply(200, { organizations: mockData.organizations });
+      })
+      .nock(cma, (api) => {
+        api.get('/v3/user?include_orgs_roles=true').reply(200, { user: mockData.users[0] });
+      })
+      .nock(cma, (api) => {
+        api.get(`/v3/organizations/${mockData.organizations[0].uid}/roles`).reply(200, { roles: mockData.roles });
+      })
+      .nock(cma, (api) => {
+        api.get('/v3/user?include_orgs_roles=true').reply(200, { user: mockData.users[0] });
+      })
```

Comment:
Both user APIs look identical. Can we add `.persist()` and have only one nock method.?

Reply:
done